syntax = "proto3";


// 不需要 date 和 url
message NewInfo {
    uint32 doc_id = 1;
    string title = 2;
    string author_photor = 3;
    string referring = 4;
    string source = 5;
    string editor = 6;
    string content = 7;
}

enum Result {
    RESULT_FAILED = 0;
    RESULT_SUCCESS = 1;
}

// 新增新闻的倒排索引的结果
message ResultResponse {
    Result result = 1;
}

service NewsOperate {
    rpc addNews(NewInfo) returns (ResultResponse);
}